jetstream/util.go

```diff
@@ -163,6 +163,17 @@ func streamConfigFromResourceData(d *schema.ResourceData) (cfg api.StreamConfig,
 		subjects[i] = sub.(string)
 	}
 
+	var subjectTransforms *api.SubjectTransformConfig
+	transforms := d.Get("subject_transform").([]any)
```

Comment:
should this really be `[]any`? `subject_transform` is a map and if it is a array due to some tf weirdness we should ensure its len is 1 

otherwise generally good but tests are failing

Reply:
Cheers, yeah it is an array with maxlength 1 according to the schema.

I've added a check for len == 1 and fixed that test, sorry should have worked out running it locally.

Reply:
It's not though, this is on the main stream:

```
    "subject_transform": {
      "description": "Subject transform to apply to matching messages",
      "type": "object",
      "required": [
        "src",
        "dest"
      ],
      "properties": {
        "src": {
          "type": "string",
          "description": "The subject transform source"
        },
        "dest": {
          "type": "string",
          "description": "The subject transform destination"
        }
      }
    },
```

and later on mirrors we have indeed an array:

```
        "subject_transforms": {
          "type": [
            "array",
            "null"
          ],
          "description": "The subject filtering sources and associated destination transforms",
          "items": {
            "minimum": 0,
            "type": "object",
            "description": "Subject transform to apply to matching messages going into the stream",
            "required": [
              "src",
              "dest"
            ],
            "properties": {
              "src": {
                "type": "string",
                "description": "The subject transform source"
              },
              "dest": {
                "type": "string",
                "description": "The subject transform destination"
              }
            }
          }
        },
```

main stream config is just a object no array

https://github.com/nats-io/jsm.go/blob/2690708e8318cd044934baa164a9afab7c81e568/api/streams.go#L554

Reply:
I was referring to this being a list https://github.com/nats-io/terraform-provider-jetstream/blob/main/jetstream/resource_jetstream_stream.go#L249

On the jsm side StreamConfig is expecting a single object like you linked though which is why I ended up with this. I'm rather new to golang though so could be misunderstanding something here.

Reply:
Yeah I think the root of the confusion is I made a mistake adding it as a lit here, it should not be a list:

https://github.com/nats-io/terraform-provider-jetstream/blob/2d699926c0a4af2c624136fbabebbc86bdac23d4/jetstream/resource_jetstream_stream.go#L249

So we have to fix that bit first and then the rest will make sense

Reply:
Hmm, that makes it a little annoying to fix as changing it from a block to an argument makes the state inconsistent.

I hit `Error: missing expected {` trying a plan against an existing state with that changed.

Reply:
hmm, ok, no choice but to keep it a map then and do the len()==1 check

Reply:
With an empty local state the same config worked, it's just when using the existing state that I hit that error.

There might be a way to convert it on the provider side to match the updated schema but I'm not familiar enough to know how to do that yet.

Reply:
ok no problem, we keep it list and take just the first one